### ISSUE TITLE
response with a JSON when an app errors out

### ISSUE THREAD

**Issue by keighrim** [clamsproject/clams-python#36]
With implementation from #33, the HTTP response body of errors is just a string-cast python error. We'd like to return a properly formatted (but not necessarily in MMIF format) JSON with error name and message, with HTTP code.

**Comment by marcverhagen** [comment]
I was wondering whether we could cannibalize the view's metadata property for this. This is partially following Kevin's suggestion that we could just return identical MMIF. I don't like that suggestion, it feels too brittle to me. But what if we set it up so that if an app raises an error, we just put that in the view.

If we run bars-and-tone without issue we get

```json
{
  "app": "http://apps.clams.ai/bars-and-tones/1.0.5",
  "timestamp": "2020-05-27T12:23:45",
  "contains": {
    "http://mmif.clams.ai/0.2.2/vocabulary/TimeFrame": {
      "unit": "seconds",
      "document": "m1" } }
}
```

Otherwise we get

```json
{
  "app": "http://apps.clams.ai/bars-and-tones/1.0.5",
  "timestamp": "2020-05-27T12:23:45",
  "error": {
    "code": "400",
    "message": "incorrect input",
    "stack-trace": "..."
  }
}
```

This view would have an empty list of annotations.

Not sure how much sense http error codes make here since we do it for the pipelining code, we may just want to put the Python error class there.

**Comment by keighrim** [comment]
I like the idea. If there was errors in input MMIF, returning the input as part of response also may help debugging. However for this, we also need to mend the json schema for views to require only one of `annotations` and `error`.

**Comment by marcverhagen** [comment]
Since `additionalProperties` is not set to false we would not need to add `error`, unless we want to make it specific, right?

**Comment by keighrim** [comment]
Yes, you're right. The view metadata does allow addition properties, so it's not a problem to throw `error` field there, I think. On other issue though is, `contains` in the viewmetadata is currently set to be one of `required` attributes. I think we can either 
1. make `viewMetadata` to require one of [`error` and `contains`]. 
2. throw `error` inside the `contains` dict. 

I'd like the first idea, as I believe that's gonna be conceptually simpler and structurally flatter. What do others think about it?

**Comment by keighrim** [comment]
#55 adds necessary code for *outputting* an error in the HTTP response. However, the `error` field in the MMIF specification is only proposed for now and yet to be merged in the stable branch (https://github.com/clamsproject/mmif/pull/166) . I will keep this issue open until the stable MMIF properly supports the `error` field in the view metadata and related code in SDK's is updated accordingly.

**Comment by marcverhagen** [comment]
I will update the MMIF specifications with the proposed changes early this week. We could choose not to wait for that because adding an error property to the view metadata won't break anything.

**Comment by keighrim** [comment]
Yep, that's what I though. I'm currently working on updating the `publish.py` file to automate some of the release actions.

**Comment by keighrim** [comment]
Closed via #57.
